downloader/src/main/kotlin/vcs/GitWorkingTree.kt

```diff
@@ -89,41 +90,55 @@ open class GitWorkingTree(workingDir: File, vcsType: VcsType) : WorkingTree(work
     }
 
     override fun getNested(): Map<String, VcsInfo> =
-        listSubmodulePaths(repo).associateWith { GitWorkingTree(repo.workTree.resolve(it), vcsType).getInfo() }
+        repo.use {
+            listSubmodulePaths(it).associateWith { path ->
+                GitWorkingTree(it.workTree.resolve(path), vcsType).getInfo()
+            }
+        }
 
     override fun getRemoteUrl(): String =
-        runCatching {
-            val remotes = org.eclipse.jgit.api.Git(repo).use { it.remoteList().call() }
-            val remoteForCurrentBranch = BranchConfig(repo.config, repo.branch).remote
-
-            val remote = if (remotes.size <= 1 || remoteForCurrentBranch == null) {
-                remotes.find { it.name == "origin" } ?: remotes.firstOrNull()
-            } else {
-                remotes.find { remote ->
-                    remote.name == remoteForCurrentBranch
+        repo.use {
+            runCatching {
+                val remotes = org.eclipse.jgit.api.Git(it).use { it.remoteList().call() }
+                val remoteForCurrentBranch = BranchConfig(it.config, it.branch).remote
+
+                val remote = if (remotes.size <= 1 || remoteForCurrentBranch == null) {
+                    remotes.find { it.name == "origin" } ?: remotes.firstOrNull()
+                } else {
+                    remotes.find { remote ->
+                        remote.name == remoteForCurrentBranch
+                    }
                 }
-            }
 
-            remote?.urIs?.firstOrNull()?.toString().orEmpty()
-        }.getOrElse {
-            throw IOException("Unable to get the remote URL.", it)
+                remote?.urIs?.firstOrNull()?.toString().orEmpty()
+            }.getOrElse {
+                throw IOException("Unable to get the remote URL.", it)
+            }
         }
 
-    override fun getRevision(): String = repo.exactRef(Constants.HEAD)?.objectId?.name().orEmpty()
+    override fun getRevision(): String = repo.use { it.exactRef(Constants.HEAD)?.objectId?.name().orEmpty() }
 
-    override fun getRootPath(): File = repo.workTree
+    override fun getRootPath(): File = repo.use { it.workTree }
 
     override fun listRemoteBranches(): List<String> =
-        runCatching {
-            LsRemoteCommand(repo).setHeads(true).call().map { it.name.removePrefix("refs/heads/") }
-        }.getOrElse {
-            throw IOException("Unable to list the remote branches.", it)
+        repo.use {
```

Comment:
The diffs of this and below function contain a couple of unrelated changes.

Reply:
Unfortunately, you're not very specific which "unrelated" changes you refer to. So once again, I can only guess that you mean the renaming of lambda arguments.

This first renaming to `branch` becomes necessary in order to not shadow the `it` from `use`. The second renaming to `e` was indeed not complained about by the IDE, but IMO it should, as keeping the original code would also shadow the outer `it`.

Reply:
I didn't notice the parameter shadowing, so I'm fine with the changes.